tokio/src/runtime/builder.rs

```diff
@@ -401,6 +409,51 @@ impl Builder {
         self
     }
 
+    /// Sets the spawn concurrency level the `Runtime` will use.
+    ///
+    /// This can be any number greater than 0 and less than or equal to 65536,
+    /// if the parameter is larger than this value, concurrency level will actually select 65536 internally.
+    ///
+    /// When the value of this is small compared to the number of concurrent threads, increasing it
+    /// will help improve the performanc of concurrently spawn tasks. However, when the value is
+    /// already large enough, further increasing it will not continue to improve performance.
+    /// Instead, it may result in longer time of the Runtime creation.
+    ///
+    /// # Default
+    ///
+    /// The default value for this is 4 times the number of worker threads.
+    ///
+    /// When using the `current_thread` runtime this method has no effect.
+    ///
+    /// # Examples
+    ///
+    /// ## Multi threaded runtime with spawn_concurrency_level 8
+    ///
+    /// ```
+    /// use tokio::runtime;
+    ///
+    /// // This will spawn a work-stealing runtime with 4 worker threads.
+    /// let rt = runtime::Builder::new_multi_thread()
+    ///     .spawn_concurrency_level(8)
+    ///     .build()
+    ///     .unwrap();
+    ///
+    /// rt.spawn(async move {});
+    /// ```
+    ///
+    /// # Panics
+    ///
+    /// This will panic if `val` is not larger than `0`.
+    #[track_caller]
+    pub fn spawn_concurrency_level(&mut self, mut val: usize) -> &mut Self {
```

Comment:
It's not clear to me that we want a configuration option for this. At least, it should be unstable.

Reply:
This is as I said above:
> The new added method `spawn_concurrency_level()` in Builder to set the number of segment lock in OwnedTasks struct is to allow us to customize this value. Its default value is 4 times the number of workers thread. An alternative approach is to set it as a constant, like 64, regardless of the number of CPU cores on different platforms. However, considering the usage of Tokio on micro-embedded devices, making the default value dependent on the number of worker threads can effectively minimize the time required for runtime creation on micro-embedded devices.

I thinke make it unstable is a good decision.